ui/src/components/admin/ArrayToExcelButton.js

```diff
@@ -0,0 +1,131 @@
+import React, { useState } from "react";
+import {
+  Button,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  Select,
+  Checkbox,
+} from "@chakra-ui/react";
+import { DownloadIcon } from "@chakra-ui/icons";
+import { arrayToExcel } from "../helpers/arrayToExcel";
+import cloneDeep from "lodash.clonedeep";
+import { useDispatch } from "react-redux";
+
+const ArrayToExcelButton = ({ apiArray, fileName, buttonTitle }) => {
+  const [showDownloadModal, setShowDownloadModal] = useState(false);
+  const [columnsType, setColumnsType] = useState("1");
+  const [selectedColumns, setSelectedColumns] = useState([]);
+  const dispatch = useDispatch();
+
+  const updateSelectedColumns = (e, column) => {
```

Comment:
not sure if there's a discussion that i am not aware of, but idk how useful it is to implement select which columns to include. Quite some work and admins can just delete the column in Excel pretty easily if they dont want it

Reply:
there was no discussion about this, i just followed a template/comprehensive guide that andrea gave me about converting array to excel. yeah true, I'll remove this function and just automatically delete password column to simplify it